sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -2893,8 +2134,11 @@ def commit_block_list( # type: ignore
         :returns: Blob-updated property dict (Etag and last modified).
         :rtype: dict(str, Any)
         """
-        options = self._commit_block_list_options(
-            block_list,
+        options = _commit_block_list_options(
+            require_encryption=self.require_encryption,
+            key_encryption_key=self.key_encryption_key,
```

Comment:
Same thing. I think you get it at this point :)

I'm going to stop commenting on this but check for more later on. Basically, if the helper function is just doing this, pull that outside and just check directly in method before calling helper.
https://github.com/Azure/azure-sdk-for-python/blob/ca79216ff9fe86b1a1b97166ec4986f4a29d572f/sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py#L1620-L1621